src/gpu/locks.rs

```diff
@@ -14,31 +15,57 @@ use crate::gpu::{CpuGpuMultiexpKernel, GpuName};
 
 const GPU_LOCK_NAME: &str = "bellman.gpu.lock";
 const PRIORITY_LOCK_NAME: &str = "bellman.priority.lock";
-fn tmp_path(filename: &str) -> PathBuf {
+const PRIORITY_LOCK_UID: &str = "00000000-0000-0000-0000-000000000000";
+
+fn tmp_path(filename: &str, id: UniqueId) -> PathBuf {
     let mut p = std::env::temp_dir();
-    p.push(filename);
+    p.push(filename.to_owned() + "." + &id.to_string());
     p
 }
 
 /// `GPULock` prevents two kernel objects to be instantiated simultaneously.
 #[allow(clippy::upper_case_acronyms)]
 #[derive(Debug)]
-pub struct GPULock(File);
-impl GPULock {
-    pub fn lock() -> GPULock {
-        let gpu_lock_file = tmp_path(GPU_LOCK_NAME);
-        debug!("Acquiring GPU lock at {:?} ...", &gpu_lock_file);
-        let f = File::create(&gpu_lock_file)
-            .unwrap_or_else(|_| panic!("Cannot create GPU lock file at {:?}", &gpu_lock_file));
-        f.lock_exclusive().unwrap();
-        debug!("GPU lock acquired!");
-        GPULock(f)
+pub struct GPULock<'a>(Vec<(File, &'a Device, PathBuf)>);
```

Comment:
Would it make sense to create a struct instead of having a tuple with 3 elements?